clusterloader2/pkg/proxypod/proxy_pod.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxypod
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+)
+
+const (
+	// ProxyPodNamespace is a proxy pod namespace.
+	ProxyPodNamespace = "measurement"
+	// ProxyPodName as a proxy pod name.
+	ProxyPodName = "proxy-pod"
+
+	proxyPodPath          = "pkg/proxypod/manifest/proxy_pod.yaml"
+	proxyPodCheckInterval = 10 * time.Second
+	proxyPodCheckTimeout  = 2 * time.Minute
+)
+
+// SetUpProxyPod creates proxy pod.
+func SetUpProxyPod(f *framework.Framework) error {
+	klog.Info("Setting up proxy pod")
+	mapping := make(map[string]interface{})
+	mapping["Namespace"] = ProxyPodNamespace
+	mapping["Name"] = ProxyPodName
+	if err := client.CreateNamespace(f.GetClientSets().GetClient(), ProxyPodNamespace); err != nil {
+		return fmt.Errorf("namespace %s creation error: %v", ProxyPodNamespace, err)
+	}
+	if err := f.ApplyTemplatedManifests(
+		proxyPodPath,
+		mapping,
+		client.Retry(apierrs.IsNotFound)); err != nil {
+		return fmt.Errorf("pod %s creation error: %v", ProxyPodName, err)
+	}
+	if err := wait.Poll(
+		proxyPodCheckInterval,
+		proxyPodCheckTimeout,
+		func() (bool, error) {
+			obj, err := f.GetObject(schema.GroupVersionKind{Version: "v1", Kind: "pod"}, ProxyPodNamespace, ProxyPodName)
```

Comment:
This is a bit strange - why we don't use typed client?

Reply:
Fixed.